src/visitor.rs

```diff
@@ -44,10 +44,7 @@ pub fn filter_inline_attrs(attrs: &[ast::Attribute], outer_span: Span) -> Vec<as
 /// Returns true for `mod foo;`, false for `mod foo { .. }`.
 fn is_mod_decl(item: &ast::Item) -> bool {
     match item.node {
-        ast::ItemKind::Mod(ref m) => {
-            !(m.inner.lo() == BytePos(0) && m.inner.hi() == BytePos(0))
-                && m.inner.hi() != item.span.hi()
-        }
+        ast::ItemKind::Mod(ref m) => m.inner.hi() != item.span.hi(),
```

Comment:
Hmm, I don't actually think this works at all - I think this condition is always false - for out of line mods the outer span ends at the `;` and the inner span ends at the end of the mod file. For inline mods the outer span includes the `}` but the inner span does not. I think we have to stick with the old method of looking at filenames, but that does mean we can't share the code for module reordering as easily.

Reply:
Actually for inline mods the inner span does include `}`. In ast.rs, it says:

```rust
/// Module declaration.
///
/// E.g. `mod foo;` or `mod foo { .. }`
#[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
pub struct Mod {
    /// A span from the first token past `{` to the last token until `}`.
    /// For `mod foo;`, the inner span ranges from the first token
    /// to the last token in the external file.
    pub inner: Span,
    pub items: Vec<P<Item>>,
}
```

Reply:
Hmm, I interpreted that comment to mean it did not include the `}`, and it is weird to include the `{` but not the `}`. However, the parser does seem to include the `}`.